### PR TITLE
Nceloss

### PR DESCRIPTION
**Pull request by safrooze**
## Description ##
Fixes and improvements to example

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage: No test exists for examples at this point
- [x] Code is well-documented: 
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Updated README with information about dataset and script behavior
- [x] Fixed scripts for python 2.7 and 3.6 conformity
- [x] Replaced Model.FeedForward with Module API
- [x] Fixed pylint issues with scripts
- [x] Removed incorrect dependency on MXNet source to be part of example tree

### CODE REVIEW

**Comment by zihaolucky** [comment]
LGTM.

File: example/nce-loss/README.md [hunk 1]
```diff
@@ -54,28 +63,26 @@ The motivation is to design a more robust and scalable word vector system, by re
 
 ### Basics
 
-<img src="https://github.com/zihaolucky/mxnet/blob/example/word2vec-nce-loss-with-subword-representations/example/nce-loss-subword-repr/slide1.png" width="700">
+<img src="images/slide1.png" width="700">
 
-<img src="https://github.com/zihaolucky/mxnet/blob/example/word2vec-nce-loss-with-subword-representations/example/nce-loss-subword-repr/slide2.png" width="700">
+<img src="images/slide2.png" width="700">
```
Comment (zihaolucky) [inline example/nce-loss/README.md]: Basically we don't put data like image into main source code, as that makes our code base too large.

File: example/nce-loss/toy_nce.py [hunk 2]
```diff
@@ -56,16 +60,16 @@ def provide_label(self):
 
 
 class DataIter(mx.io.DataIter):
-    def __init__(self, count, batch_size, vocab_size, num_label, feature_size):
+    def __init__(self, count, batch_size_, vocab_size_, num_label_, feature_size_):
         super(DataIter, self).__init__()
-        self.batch_size = batch_size
+        self.batch_size = batch_size_
```
Comment (eric-haibin-lin) [inline example/nce-loss/toy_nce.py]: Does trailing underscore follow google's style guide for python?
https://google.github.io/styleguide/pyguide.html#Naming

Comment (safrooze) [reply example/nce-loss/toy_nce.py]: Trailing underscores for name clashes are part of PEP8: https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments
The reason for the change is that variable names in the function clash with variable names in the outer scope.

File: example/nce-loss/README.md (hunk 1 above)
Comment (safrooze) [reply example/nce-loss/README.md]: Sure. But linking to images from an unknown repo can break the README in the future. Also I followed this example where the image is part of the repo: https://github.com/apache/incubator-mxnet/tree/master/example/captcha
Is linking to the image in a personal repo preferred?!

Comment (safrooze) [reply example/nce-loss/README.md]: Alternatively I can just remove the images and leave it to user to look up the paper in the link.

**Comment by piiswrong** [comment]
we don't include binaries (images) in the repo. Put them at some other place (dmlc/web-data) and link to them

**Comment by safrooze** [comment]
- Will remove images
- Will move functions into a separate file to avoid name clash

**Pull request closed without merging**